src/webapp/src/characters/charactersSlice.ts

```diff
@@ -0,0 +1,63 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+interface StateData {
+  results: StateCharacter[];
+  error?: string | null;
+}
+
+export interface StateCharacter {
+  id: number;
+  name: string;
+  species: string;
+  image: string;
+}
+interface ApiKnownError {
+  error: string;
+}
+const initialState = {
+  results: [] as StateCharacter[],
+  error: null,
+} as StateData;
+
+export const getCharacters = createAsyncThunk<
+  StateCharacter[],
+  void,
+  { rejectValue: ApiKnownError }
+>('characters/getCharacters', async (_, { rejectWithValue }) => {
+  const response = await fetch("https://rickandmortyapi.com/api/character");
+
+  if (response.status === 404) {
+    return rejectWithValue((await response.json()) as ApiKnownError);
+  }
+  const { results } = await response.json();
+
+  return results.map(({ species, image, id, name }: StateCharacter) => ({
+    species,
+    image,
+    id,
+    name,
+  }));
+});
```

Comment:
I didn't see this one either. Two comment were missed somehow. I mean, they were not there... Is that possible? :thinking: Maybe it didn't refresh? Hmm... I have to figure that out. I wouldn't like to miss thinks by mistake.

Reply:
I thought one of the ideas of `redux-toolkit` was to put everything concerning one slice in the same file. With the new way of creating actions we don't need an actions file, or not? In the end we don't need to use the `createAction` or the `createReducer` functions, we can do it everything with `createSlice`

I thought that would be good because I saw like that in the `redux-toolkit` documentation. 

BTW, I think there are still missing comments. I think when I was at the dentist I saw you mentioned something about the file name for the actions, but I cannot find it anymore.